frontend/storage/utils/downloadGameData.ts

```diff
@@ -0,0 +1,57 @@
+import { AnyAction, Dispatch } from 'redux';
+import { AppActions, SERVER_ADRESS, DataLoadActions, BreadcrumbActions } from '../common';
+import axios, { AxiosResponse } from 'axios';
+import { GameSeries } from '../models/GameSeries';
+import { Game } from '../models/Game';
+import { Image } from '../models/Image';
+import { GameEnvironment } from '../models/GameEnvironment';
+import { Environment } from '../models/Environment';
+
+export function requestGameData(id: number, dispatch: Dispatch<AnyAction>, changeGameExistsState: (gameFound: boolean) => void) {
+	fetchGameData(id)(dispatch, changeGameExistsState);
+}
+function fetchGameData(id: number) {
+	return (dispatch: Dispatch<AppActions>, changeGameExistsState: (gameFound: boolean) => void) => {
```

Comment:
If you rewrite it to be `async` method you can simplify things greatly and avoid the nesting. See how I did it in the backend.